shell/browser/atom_browser_client.cc

```diff
@@ -1045,28 +1065,130 @@ void AtomBrowserClient::RegisterNonNetworkNavigationURLLoaderFactories(
     protocol->RegisterURLLoaderFactories(factories);
 }
 
+namespace {
+
+// The FileURLLoaderFactory provided to the extension background pages.
+// Checks with the ChildProcessSecurityPolicy to validate the file access.
+class FileURLLoaderFactory : public network::mojom::URLLoaderFactory {
```

Comment:
This might not be required actually? TODO: try removing it and see if the pdf extension still works